commit 1e746fe215eb0e37ff1d91bb894a410e2eb05796
Author: Liviu Dudau <liviu@dudau.co.uk>
Date:   Sat Dec 29 17:23:43 2018 +0000

    nvme-pci: use the same attributes when freeing host_mem_desc_bufs.
    
    [ Upstream commit cc667f6d5de023ee131e96bb88e5cddca23272bd ]
    
    When using HMB the PCIe host driver allocates host_mem_desc_bufs using
    dma_alloc_attrs() but frees them using dma_free_coherent(). Use the
    correct dma_free_attrs() function to free the buffers.
    
    Signed-off-by: Liviu Dudau <liviu@dudau.co.uk>
    Signed-off-by: Christoph Hellwig <hch@lst.de>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/nvme/host/pci.c b/drivers/nvme/host/pci.c
index d668682f91df..40f76b4f08fc 100644
--- a/drivers/nvme/host/pci.c
+++ b/drivers/nvme/host/pci.c
@@ -1727,8 +1727,9 @@ static void nvme_free_host_mem(struct nvme_dev *dev)
 		struct nvme_host_mem_buf_desc *desc = &dev->host_mem_descs[i];
 		size_t size = le32_to_cpu(desc->size) * dev->ctrl.page_size;
 
-		dma_free_coherent(dev->dev, size, dev->host_mem_desc_bufs[i],
-				le64_to_cpu(desc->addr));
+		dma_free_attrs(dev->dev, size, dev->host_mem_desc_bufs[i],
+			       le64_to_cpu(desc->addr),
+			       DMA_ATTR_NO_KERNEL_MAPPING | DMA_ATTR_NO_WARN);
 	}
 
 	kfree(dev->host_mem_desc_bufs);
@@ -1794,8 +1795,9 @@ static int __nvme_alloc_host_mem(struct nvme_dev *dev, u64 preferred,
 	while (--i >= 0) {
 		size_t size = le32_to_cpu(descs[i].size) * dev->ctrl.page_size;
 
-		dma_free_coherent(dev->dev, size, bufs[i],
-				le64_to_cpu(descs[i].addr));
+		dma_free_attrs(dev->dev, size, bufs[i],
+			       le64_to_cpu(descs[i].addr),
+			       DMA_ATTR_NO_KERNEL_MAPPING | DMA_ATTR_NO_WARN);
 	}
 
 	kfree(bufs);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1388_1e746fe215eb0e37ff1d91bb894a410e2eb05796.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1388_1e746fe215eb0e37ff1d91bb894a410e2eb05796.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
